build/pipelines/templates/build-console-steps.yml

```diff
@@ -32,6 +32,29 @@ steps:
     restoreSolution: build/packages.config
     restoreDirectory: '$(Build.SourcesDirectory)\packages'
 
+# The environment variable VCToolsInstallDir isn't defined on lab machines, so we need to retrieve it ourselves.
+- script: |
+    "%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -requires Microsoft.Component.MSBuild -property InstallationPath > %TEMP%\vsinstalldir.txt
+    set /p _VSINSTALLDIR15=<%TEMP%\vsinstalldir.txt
+    del %TEMP%\vsinstalldir.txt
+    call "%_VSINSTALLDIR15%\Common7\Tools\VsDevCmd.bat"
+    echo VCToolsInstallDir = %VCToolsInstallDir%
+    echo ##vso[task.setvariable variable=VCToolsInstallDir]%VCToolsInstallDir%
+  displayName: 'Retrieve VC tools directory'
+
+- task: CmdLine@1
+  displayName: 'Display build machine environment variables'
+  inputs:
+    filename: 'set'

```

Comment:
no true need

Reply:
True but it makes it so much easier to see what's going on so I'd prefer to leave it.